infra/modules/aws-vault-auth/main.tf

```diff
@@ -0,0 +1,81 @@
+# provisions User, key, AWS IAM Policy needed for AWS IAM Auth method in Vault; and the vault backend
+# **ALPHA support**
+
+# AWS IAM user for Vault
+resource "aws_iam_user" "vault" {
+  name = "VaultAwsAuth"
+}
+
+# from https://developer.hashicorp.com/vault/docs/auth/aws#recommended-vault-iam-policy
+resource "aws_iam_user_policy" "vault" {
+  name = "VaultAwsAuthPolicy"
+  user = aws_iam_user.vault.name
+
+  policy = jsonencode(
+    {
+      "Version" : "2012-10-17",
+      "Statement" : [
+        {
+          "Effect" : "Allow",
+          "Action" : [
+            "ec2:DescribeInstances",
+            "iam:GetInstanceProfile",
+            "iam:GetUser",
+            "iam:GetRole"
+          ],
+          "Resource" : "*"
+        },
+        {
+          "Effect" : "Allow",
+          "Action" : [
+            "sts:AssumeRole"
+          ],
+          "Resource" : [
+            "arn:aws:iam::874171213677:role/VaultRole"
```

Comment:
:yellow_circle: Which ARN is 874171213677 for? I'd parametrize it and/or explain what it is

Reply:
yeah, mistake; the whole ARN should be a parameter - thanks!